Microsoft.Research/Contracts/System.Core.Contracts/System.Linq.Expressions.SwitchCase.cs

```diff
@@ -44,8 +45,11 @@ sealed public partial class SwitchCase
     #region Methods and constructors
     private SwitchCase() {}
 
+    [Pure]
     public SwitchCase Update (IEnumerable<Expression> testValues, Expression body)
     {
+      Contract.Requires(body != null);
```

Comment:
@sharwell again. We can't add this as a precondition. `null` is an absolutely valid value. Please look here: http://referencesource.microsoft.com/#System.Core/Microsoft/Scripting/Ast/SwitchCase.cs,100